src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/TypeScript/components/App.tsx

```diff
@@ -90,6 +66,22 @@ function App() {
   const toggleSettings = () => setIsSettingsOpen(!isSettingsOpen);
   const closeSettings = () => setIsSettingsOpen(false);
 
+  const downloadDataset = () => {
+    // create a stringified JSON of the dataset
+    const dataStr = JSON.stringify(dataset, null, 2);
+
+    // create a link to download the JSON file in the page and click it
+    const blob = new Blob([dataStr], { type: 'application/json' });
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = `${scoreSummary.primaryResult.executionName}.json`;
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+    URL.revokeObjectURL(url);
```

Comment:
The DOM manipulation seems a bit hacky - but I guess there isn't a more direct way to accomplish this in react?

Reply:
That's the only way to do it from an SPA page. There might a component that wraps it, but that doesn't seem necessary.